### ISSUE TITLE
Separate initialization from sampling in templates.QSO

### ISSUE THREAD

**Issue by dkirkby** [desihub/desisim#48]
The class constructor reads the basis templates from disk but also takes the number of templates to generate as a parameter.  This means that the obvious implementation of a loop that generates a batch of QSOs in each iteration ends up re-reading the basis templates each time:
```
for i in xrange(1000):
   qso = QSO(model=100) # reads qso_templates_v1.1.fits
   outflux, wave, meta = qso.make_templates()
```
I suspect the fix is as simple as moving the `nmodel` parameter from the ctor to the `make_templates` method, leading to:
```
qso = QSO() # reads qso_templates_v1.1.fits
for i in xrange(1000):
   outflux, wave, meta = qso.make_templates(nmodel=100)
```
Btw, the `make_templates` docstring does not describe the `wave` element of the returned tuple.

**Comment by dkirkby** [comment]
I suggest also moving the `seed` arg from the ctor to `make_templates`.  In any case, the code does not currently generate identical templates when the same seed is repeated, but it looks like replacing:
```
this = np.random.random_integers(0,nbase)
```
with
```
this = self.rand.random_integers(0,nbase)
```
would fix this.

**Comment by dkirkby** [comment]
Actually, that line should be:
```
this = self.rand.random_integers(0, nbase - 1)
```
since the upper limit passed to `random_integers` is inclusive.

**Comment by sbailey** [comment]
Fixed in #49, including tests that using the same seed produces the same results and using different seeds produce different results.
